### PR TITLE
Feature/player attack

### PR DESCRIPTION
**Pull request by Toine2106**
First version of player attack
Enemy controller with health and gets destroyed when it hits 0

### CODE REVIEW

File: Assets/Scripts/Statemachine/States/PlayerIdleState.cs [hunk 1]
```diff
@@ -1,20 +1,34 @@
 using UnityEngine;
 
 public class PlayerIdleState : BaseState {
+
+	private bool _attack = false;
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerIdleState.cs:5]: Isn't it better to store this in the PlayerController?

File: Assets/Scripts/Statemachine/States/PlayerMoveState.cs [hunk 2]
```diff
@@ -4,13 +4,15 @@
 public class PlayerMoveState : BaseState {
 
 	bool _dash = false;
+	bool _attack = false;
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerMoveState.cs:7]: Once again, isn't this better stored in the PlayerController?

**Comment by HIHIQY1** [comment]
I'm getting errors for PlayerAttackState.cs. Please use a cast or ensure the types are correct in the base class (e.g. by using generics)

afbeelding

File: Assets/Scripts/Statemachine/States/PlayerAttackState.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+
+public class PlayerAttackState : BaseState {
+	float _speed = 10;
+	private bool _onAttack = true;
+	public PlayerAttackState(string name, StateMachine stateMachine) : base(name, stateMachine) { }
+	public override void EnterState() {
+
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerAttackState.cs:11]: Empy line

File: Assets/Scripts/Statemachine/States/PlayerAttackState.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	public override void UpdateState() {
+		StateMachine.transform.Translate(_speed * Time.deltaTime * Movement);
+		if (!_onAttack) {
+			StateMachine.SwitchState("Idle");
+		}
+		else if (Movement == Vector2.zero) {
+			StateMachine.SwitchState("Move");
+		}
+
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerAttackState.cs:24]: Empy line as well

File: Assets/Scripts/Statemachine/States/PlayerDashState.cs [hunk 5]
```diff
@@ -5,19 +5,16 @@
...
 	public override void EnterState() {
-		// StateMachine.StartCoroutine(dashCooldown());
 		StateMachine.StartCoroutine(dashTime());
 	}
 
 	public override void UpdateState() {
-		if (_dash) {
-			StateMachine.transform.Translate(Movement * _dashSpeed * Time.deltaTime);
-		}
+
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerDashState.cs:16]: Empy line

File: Assets/Scripts/Statemachine/States/PlayerMoveState.cs [hunk 6]
```diff
@@ -4,13 +4,15 @@
...
 	float _speed = 10;
 	bool _dashCooldown = false;
 	public PlayerMoveState(string name, StateMachine stateMachine) : base(name, stateMachine) {
 	}
 
 	public override void EnterState() {
 		EventBus.Instance.Subscribe<bool>(EventType.DASH, OnDash);
+		EventBus.Instance.Subscribe<bool>(EventType.ATTACK, OnAttack);
```
Comment (HIHIQY1) [inline Assets/Scripts/Statemachine/States/PlayerMoveState.cs:16]: Empy line

File: Assets/Scripts/Statemachine/States/PlayerIdleState.cs (hunk 1 above)
Comment (Toine2106) [reply Assets/Scripts/Statemachine/States/PlayerIdleState.cs:5]: can be implemented later on

**Pull request merged**